Startup/FullTime/ConfessionDay.js

```diff
@@ -1,78 +1,89 @@
-    }
-
-//Only continue if we haven't check this today
-    if (!isVar(VARIABLE_NEXT_CONFESSION_DAY) || getDate(VARIABLE_NEXT_CONFESSION_DAY).hasPassed()) {
-        if (!getDate(VARIABLE_NEXT_CONFESSION_DAY).sameDay(setDate())) {
-            sendVirtualAssistantMessage("You've been skipping confession day %SlaveName%!");
-            sendVirtualAssistantMessage("I've added a healthy dose of punishment points for that!");
-            addPunishmentPoints(300);
-            sendVirtualAssistantMessage("You are expected to report on tuesdays!");
-        }
-
-        //Set the date to the next tuesday
-        setDate(VARIABLE_NEXT_CONFESSION_DAY, setDate(VARIABLE_NEXT_CONFESSION_DAY).addDay(7).setHour(0).setMinute(0).setSecond(0));
-
-        sendVirtualAssistantMessage("Now slave tell me...");
-        sendVirtualAssistantMessage("Confess!", false);
-        while (true) {
-            let answer = createInput();
-
-            if (answer.containsIgnoreCase("came, orgasm, cum, cumming")) {
-                sendVirtualAssistantMessage("Cumming without permission...");
-                addPunishmentPoints(400);
-            } else if (answer.containsIgnoreCase("ruin", "ruined")) {
-                sendVirtualAssistantMessage("Ruining without permission...");
-                addPunishmentPoints(250);
-            } else if (answer.containsIgnoreCase("stroke", "stroking", "touched", "pleasured")) {
-                sendVirtualAssistantMessage("Stroking without permission...");
-                addPunishmentPoints(150);
-            } else if (answer.containsIgnoreCase("lazy", "sloppy")) {
-                sendVirtualAssistantMessage("Lazy behaviour...");
-                addPunishmentPoints(150);
-            } else if (answer.containsIgnoreCase("unhealthy")) {
-                sendVirtualAssistantMessage("Eating unhealthy...");
-                addPunishmentPoints(150);
-            } else {
-                sendVirtualAssistantMessage("This isn't good slave!");
-                addPunishmentPoints(150);
-            }
-
-            sendVirtualAssistantMessage("Anything else?", false);
-
-            let more = false;
-            answer = createInput();
-
-            while (true) {
-                if (answer.isLike("yes")) {
-                    sendVirtualAssistantMessage("Tell me %SlaveName%!");
-                    more = true;
-                    break;
-                } else if (answer.isLike("no")) {
-                    sendVirtualAssistantMessage("Good %SlaveName%!");
-                    break;
-                } else {
-                    sendVirtualAssistantMessage(YES_OR_NO);
-                    answer.loop();
-                }
-            }
-
-            if (!more) {
-                break;
-            }
-        }
-
-        sendVirtualAssistantMessage("I've added punishment points matching the crime(s)!");
-        sendVirtualAssistantMessage("Make sure you report for punishment soon!");
-    }
+{

```

Comment:
Good idea but I guess we should change the system of tracking what happened to something more flexible. I will think about it

Reply:
yeah  (this was also pretty old code.....  I should definitely try and parse the input and set the punishment reason flags....and maybe come up with more specific responses)